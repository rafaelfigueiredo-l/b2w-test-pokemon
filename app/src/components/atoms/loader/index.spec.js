import Loader from ".";

describe("<Loader />", () => {
  const image = require("../../../assets/images/loader.gif");

  const component = <Loader alt="AltText" src={image} width="50px" />;

  const mounted = mount(component);
  const wrapper = shallow(component);

  it("should mount correctly Loader component", () => {
    expect(wrapper.find(".loader")).toHaveLength(1);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly Loader component", () => {
    expect(mounted.unmount()).toEqual({});
  });

  it("should render correctly alt property", () => {
    expect(wrapper.prop("alt")).toEqual("Loader");
  });

  it("should render correctly src property", () => {
    expect(wrapper.prop("src")).toEqual(image);
  });

  it("should render correctly width property", () => {
    expect(wrapper.prop("width")).toEqual("50px");
  });
});
