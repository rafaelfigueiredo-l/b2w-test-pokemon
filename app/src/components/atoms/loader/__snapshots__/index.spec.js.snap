// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should mount correctly Loader component 1`] = `
<styled.img
  alt="Loader"
  className="loader"
  src="loader.gif"
  width="50px"
>
  <img
    alt="Loader"
    className="sc-AxjAm euZXpc loader"
    src="loader.gif"
    width="50px"
  />
</styled.img>
`;
